jumanji/environments/combinatorial/binpack/env.py

```diff
@@ -156,6 +159,46 @@ def __repr__(self) -> str:
             ]
         )
 
+    @classmethod
+    def create_instance_generator(
+        cls, instance_generator: str, **instance_generator_kwargs: Any
+    ) -> InstanceGenerator:
+        """
+        Factory method for creating an instance generator.
+
+        This method can be overridden to add new instance generator types.
+
+        Args:
+            instance_generator: The type of instance generator to create. Possible values:
+                - 'toy': Create a toy instance generator.
+                - 'csv': Create a CSV instance generator.
+                - 'random': Create a random instance generator.
+            **instance_generator_kwargs:
+                Additional keyword arguments to pass to the instance generator constructor.
+
+        Returns:
+            An instance of `InstanceGenerator`.
+
+        Raises:
+            ValueError: If an unexpected value is provided for `instance_generator`.
+        """
+        instance_generator_obj: InstanceGenerator
+
+        if instance_generator == "toy":
+            instance_generator_obj = ToyInstanceGenerator()
+        elif instance_generator == "csv":
+            instance_generator_obj = CSVInstanceGenerator(**instance_generator_kwargs)
+        elif instance_generator == "random":
+            instance_generator_obj = RandomInstanceGenerator(
+                **instance_generator_kwargs
+            )
+        else:
+            raise ValueError(
+                f"Unexpected value for 'instance_generator', got {instance_generator!r}."
+                "Possible values: 'toy', 'csv', 'random'."
+            )
+        return instance_generator_obj
```

Comment:
Small suggestion to factorise this a little (not sure about first-line typing).
```suggestion
        instance_generator_cls: Type[InstanceGenerator]

        if instance_generator == "toy":
            instance_generator_cls = ToyInstanceGenerator
        elif instance_generator == "csv":
            instance_generator_cls = CSVInstanceGenerator
        elif instance_generator == "random":
            instance_generator_cls = RandomInstanceGenerator
        else:
            raise ValueError(
                f"Unexpected value for 'instance_generator', got {instance_generator!r}."
                "Possible values: 'toy', 'csv', 'random'."
            )
        return instance_generator_cls(**instance_generator_kwargs)
```

Since this is essentially a factory method, we could alternatively use a dictionary property
```
self.instance_generator_factory = {
    "toy": ToyInstanceGenerator,
    "csv": CSVInstanceGenerator,
    "random": RandomInstanceGenerator,
}
```
But this is essentially the same, what do you think?

Reply:
Thanks for the suggestions. 

The `ToyInstanceGenerator` class doesn't take any parameter so linters won't allow to pass (even empty) kwargs because of that.

Regarding the dictionary vs class method -  I lean towards the class method one. It's closer to the "strategy" design pattern, uses the OOP principles, is a bit more explicit and 'immutable'.

Reply:
I see, thanks!
@djbyrne what do you think is preferred in this case? A more factorised code and a type ignore or the initial solution?

Reply:
TBH, I am not a fan of type ignores if we can avoid them. So only based on that I would lean toward just instantiating the objects using the initial solution. 

@clement-bonnet  I do still like the idea of having a directory of classes that is used to instantiate the object. I think it is still following the factory design pattern as it is creational as opposed to operational and would argue that it is still following OOP principles if it wasn't for the linting error.